akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -276,7 +281,7 @@ class IslandTracking(
         } else {
           if (Debug) println(s"    cross-island wiring to local slot $localInSlot in target island")
           val publisher = currentPhase.createPublisher(out, logic)
-          targetSegment.phase.takePublisher(localInSlot, publisher)
+          targetSegment.phase.takePublisher(localInSlot, publisher, logic)
```

Comment:
This is wrong. We are wiring back to a module we already visited. This logic you pass here is not that logic. And now I remember why it does not take a `logic` as parameter and why it is not passed in L240. If the island needs the logic it has to keep track of it itself, but it should not impose the tracking cost to islands that don't need it.